### PR TITLE
[flang] Update the date_and_time intrinsic for AIX

### PR DESCRIPTION
**Pull request by kkwli**
Currently, the `strftime` routine is called to get the timezone for the `ZONE` argument. On AIX, this routine requires an environment variable (i.e. `XPG_SUS_ENV=ON`) set in order to return _+HHMM_/_-HHMM_ with the `%z` format. This patch is to add computation of the time difference from UTC for AIX platform.

### CODE REVIEW

**Comment by kkwli** [comment]
Ping

**Review by jeanPerier — COMMENTED** [review]
Small style nits. I am not an expert of time intrinsics and cannot test the patch on AIX though.

File: flang/runtime/time-intrinsic.cpp [hunk 1]
```diff
@@ -263,8 +316,13 @@ GetGmtOffset(const TM &tm, fallback_implementation) {
   // tm.tm_gmtoff is not available, there may be platform dependent alternatives
   // (such as using timezone from <time.h> when available), but so far just
   // return -HUGE to report that this information is not available.
-  return -std::numeric_limits<Fortran::runtime::CppTypeFor<
-      Fortran::common::TypeCategory::Integer, KIND>>::max();
+  bool err{false};
+  auto diff{computeUTCDiff(tm, &err)};
```
[the remark this answers is not in the document]
Comment (kkwli) [reply flang/runtime/time-intrinsic.cpp:320]: I will guard it.

**Review by DanielCChen — APPROVED** [review]
LGTM.

**Comment by kkwli** [comment]
Thanks for the review.

**Pull request merged**